### ISSUE TITLE
Open Startify on new tabs

### ISSUE THREAD

**Issue by jerzygangi** [mhinz/vim-startify#139]
I'm trying to launch Startify every time a new tab opens in vim.

I've tried hacking together a solution, copying parts of vim-startify/plugin/startify.vim into my .vimrc -- but to no avail.

How can this be accomplished?

**Comment by mhinz** [comment]
@fmoralesc Apparently it wasn't merged. Sorry for that, but I know that feeling. :\

@jerzygangi Can you live with the snippet given above?

**Comment by fmoralesc** [comment]
@mhinz To be fair, I slacked off and didn't finish the WinEnter autocommands :/

**Comment by mhinz** [comment]
Keep nagging Bram about it! It seems there was no final decision but it "just gone under".

**Comment by fmoralesc** [comment]
I think the suggestion to implement the `WinEnter` autocmds brought me down, because I couldn't make them behave well. For now, I'll just focus on the Tab related events.

**Comment by mhinz** [comment]
Yup, don't overwhelm them. :)

**Comment by fmoralesc** [comment]
@mhinz I just updated the patches for vim: https://groups.google.com/forum/#!topic/vim_dev/bduueOtT-4c

Btw, what happened with https://github.com/neovim/neovim/pull/1216 and https://github.com/neovim/neovim/pull/1221? It seems like EndOfBuffer is outdated at neovim.

**Comment by mhinz** [comment]
The `EndOfBuffer` patch just needs an update. Do you want to do it? Otherwise I'll look into it again.

The the other patch.. is apparently quite controversial. So.. I'm not sure what to expect.

Since I sent both patches upstream afterwards, it looks as if they waited for them to be merged into Vim. This way there would be nothing to argue anymore, since NeoVim tries to stay as compatible as possible, and the patches could be backported.

..but since both patche landed in Bram's bottomless pit that is "the todo list", I doubt that they will be merged in the near future.

**Comment by fmoralesc** [comment]
I think it's best if you updated the `EndOfBuffer` yourself (mainly not to split the feature history), but if you don't have time I wouldn't mind to propose the updated patch. @justinmk seemed open to merging it a month ago, but was waiting on the update. I also don't see any compatibility issue, this is basically equivalent to adding a new highlight group in a syntax file, and linking to NonText by default makes it behave the same as before.

As for the other patch, I don't know either. @noahfrederick might be right in that the best behaviour would be to make "all columns, including the number column, extend to the bottom of the window."

**Comment by mhinz** [comment]
It would be nice if you would take over. As far as I see it, you can just use the patch I sent upstream (the one justinmk links to) and add project-specific stuff like doxygen comments etc. It's a simpler as it just uses a "default link" instead of patching each standard colorscheme.

And the other patch... meh. :-) I used to sport it in my local branch, but don't care enough about it anymore. The community has to decide.

**Comment by fmoralesc** [comment]
Ok, then, I'll take over that. ;)

**Comment by fmoralesc** [comment]
See, it got merged in neovim ;)

**Comment by mhinz** [comment]
@fmoralesc Well done. :-)

@jerzygangi Ping?

**Comment by jerzygangi** [comment]
@mhinz Sorry for the delayed reply. I gave up on vim-startify a while ago, because I wasn't able to get it to open in new tabs consistently.

**Comment by mhinz** [comment]
Okay, I guess there's nothing one can do about it unless Vim supports that feature.

**Comment by fmoralesc** [comment]
@mhinz Just as an update: neovim merged the `TabNew` patches (yay!), so there at least this can be implemented as 
~~~ vim
if has('nvim')
    au! TabNewEntered * Startify
endif
~~~

**Comment by mhinz** [comment]
Woah, pretty cool. Grats. :-)

**Comment by jerzygangi** [comment]
Nice work @fmoralesc !!

**Comment by kami-zh** [comment]
FYI, I don't use neovim but I can open Satrtify on new tabs by following mapping:

```
nnoremap <Leader>t :tabnew<CR>:Startify<CR>
```

**Comment by fmoralesc** [comment]
@kami-zh Vim also has TabNew these days, Bram took the idea.

**Comment by for-coursera** [comment]
@fmoralesc Does something like this work for you in Vim8?

    au! TabNew * Startify

Somehow it doesn't for me (yet `au! TabNew * echom "yaaaay"` does)

**Comment by mhinz** [comment]
@for-coursera 

Unfortunately `TabNew` triggers _too early_. `TabNewEntered` is needed. Neovim supports both, Vim only the former.

Simple test: `au! TabNew * let b:foo = 42`. After `:tabnew`, `:echo b:foo` won't work, because it was set in the _previous_ buffer.

In the case of `:au! TabNew * Startify | sleep 3`, Startify actually is run, but in a weird state somewhere between the old and new buffer, and after the `:sleep` is over, Vim will empty the buffer, because it always does at the end of `:tabnew`.

**Comment by for-coursera** [comment]
@mhinz, Thanks for the info! So, as of now, we still stick to something like this, right?

    autocmd! BufEnter * if !exists('t:new_tab') && expand('%') == "" | let t:new_tab = 1 | Startify | endif

**Comment by mhinz** [comment]
Yup, looks good to me. I guess I should add that to `:h startify-faq`.

**Comment by mhinz** [comment]
Or wait, `vim foo`, `:h :help` fails for me, because `t:new_tab` isn't set in that case.

What about this?

```vim
if has('nvim')
  autocmd TabNewEntered * Startify
else
  autocmd VimEnter * let t:startify_new_tab = 1
  autocmd BufEnter *
        \ if !exists('t:startify_new_tab') && empty(expand('%')) |
        \   let t:startify_new_tab = 1 |
        \   Startify |
        \ endif
endif
```

**Comment by for-coursera** [comment]
Yeah, confirm that fail, too. And adding `autocmd VimEnter * let t:startify_new_tab = 1` indeed solves the case.

Thanks for that hint! :)
